pandas/io/parsers/base_parser.py

```diff
@@ -745,8 +745,9 @@ def _cast_types(self, values: ArrayLike, cast_type: DtypeObj, column) -> ArrayLi
                 if isinstance(cast_type, BooleanDtype):
                     # error: Unexpected keyword argument "true_values" for
                     # "_from_sequence_of_strings" of "ExtensionArray"
+                    values_str = [str(val) for val in values]
                     return array_type._from_sequence_of_strings(  # type: ignore[call-arg]
-                        values,
+                        values_str,
                         dtype=cast_type,
                         true_values=self.true_values,
                         false_values=self.false_values,
```

Comment:
`none_values` is not passed anything here. Can the equivalent `na_values` and `keep_default_na` be passed in here, or is it irrelevant?

Reply:
I think I see what you're saying. You're suggesting to propagate `na_values` and `keep_default_na` from read_excel() to _cast_types()

Reply:
Those params are already available in _cast_types since it's an instance method (so self.na_values and self.keep_default_na are available here, similar to self.true_values and self.false_values). I was thinking about progagating them to `_from_sequence_of_strings()`

Reply:
Yep we're on the same page. Making those changes now